spinalcordtoolbox/scripts/sct_analyze_lesion.py

```diff
@@ -250,13 +248,13 @@ def _measure_volume(self, im_data, p_lst, idx):
         printv('  Volume : ' + str(np.round(vol_tot_cur, 2)) + ' mm^3', self.verbose, type='info')
 
     def _measure_length(self, im_data, p_lst, idx):
-        length_cur = np.sum([np.cos(self.angles[zz]) * p_lst[2] for zz in np.unique(np.where(im_data)[2])])
+        length_cur = np.sum([p_lst[2] / np.cos(self.angles[zz]) for zz in np.unique(np.where(im_data)[2])])
         self.measure_pd.loc[idx, 'length [mm]'] = length_cur
         printv('  (S-I) length : ' + str(np.round(length_cur, 2)) + ' mm', self.verbose, type='info')
 
     def _measure_diameter(self, im_data, p_lst, idx):
         area_lst = [np.sum(im_data[:, :, zz]) * np.cos(self.angles[zz]) * p_lst[0] * p_lst[1] for zz in range(im_data.shape[2])]
```

Comment:
Took me a little while to wrap my head around the angle correction stuff. Had to visualize? Especially to understand why the angle was originally computed via `arccos(tangent_vector <dot> unit_z_vector)`.

![Screenshot from 2022-02-11 11-29-44](https://user-images.githubusercontent.com/16181459/153630341-a7febc1d-67aa-4735-9883-48c3c80d85ca.png)

But this makes sense to me now, I think. I can also adjust the text to confirm this behavior, too.

Reply:
Done in [1908f1a](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3681/commits/1908f1a2fe226567e709f5900fb601c8a0b4cd55).